minari/dataset/_storages/parquet_storage.py

```diff
@@ -0,0 +1,295 @@
+        )
+
+    elif is_image_space(space):
+        first_image = io.BytesIO((values[0]).as_py())
+        first_image = np.array(Image.open(first_image))
+        jpeg_images = np.empty((len(values),) + first_image.shape, dtype=np.uint8)
+        jpeg_images[0] = first_image
+        for i, jpeg_bytes in enumerate(values[1:], start=1):
+            jpeg_bytes = io.BytesIO(jpeg_bytes.as_py())
+            jpeg_images[i] = np.array(Image.open(jpeg_bytes))
+        return jpeg_images
+
+    elif isinstance(space, _FIXEDLIST_SPACES):
+        data = np.stack(values.to_numpy(zero_copy_only=False))
+        return data.reshape(-1, *space.shape)
+    elif isinstance(space, gym.spaces.Discrete):
+        return values.to_numpy()
+    else:
+        return values.to_pylist()
+
+
+def _encode_info(info: dict):
+    arrays, fields = [], []
+
+    for key, values in info.items():
+        if isinstance(values, dict):
+            array = _encode_info(values)
+            arrays.append(array)
+            fields.append(pa.field(key, array.type))
+
+        elif isinstance(values, tuple):
+            array = _encode_info({str(i): v for i, v in enumerate(values)})
+            arrays.append(array)
+            fields.append(pa.field(key, array.type))
+
+        elif isinstance(values, np.ndarray) or (
+            isinstance(values, Sequence) and isinstance(values[0], np.ndarray)
+        ):
+            if isinstance(values, Sequence):
+                values = np.stack(values)
+
+            data_enc = ",".join(str(n) for n in values.shape[1:])
+            values = values.reshape(len(values), -1)
+            dtype = pa.from_numpy_dtype(values.dtype)
+            struct = pa.list_(dtype, list_size=values.shape[1])
+            arrays.append(
+                pa.FixedSizeListArray.from_arrays(values.reshape(-1), type=struct)
+            )
+            fields.append(pa.field(key, struct, metadata={"shape": data_enc}))
+
+        else:
+            array = pa.array(list(values))
+            arrays.append(array)
+            fields.append(pa.field(key, array.type))
+
+    return pa.StructArray.from_arrays(arrays, fields=fields)
+
+
+def _decode_info(values: pa.Array):
+    nested_dict = {}
+    for i, field in enumerate(values.type):
+        if isinstance(field, pa.StructArray):
+            nested_dict[field.name] = _decode_info(values.field(i))
+        else:
+            value = np.stack(values.field(i).to_numpy(zero_copy_only=False))
+            if field.metadata is not None and b"shape" in field.metadata:
+                data_shape = field.metadata[b"shape"].split(b",")
+                data_shape = tuple(int(d) for d in data_shape)
+                value = value.reshape(len(value), *data_shape)
+            nested_dict[field.name] = value
+    return nested_dict
+
+
+class NumpyEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+        return super().default(obj)
```

Comment:
I guess this file is mostly a copy paste of arrow_storage.py, isn't it?

I would subclass arrow_storage instead, overriding the FORMAT attribute.